easybuild/framework/easyblock.py

```diff
@@ -1566,6 +1570,24 @@ def sanity_check_step(self, custom_paths=None, custom_commands=None, extension=F
         else:
             self.log.info("Using specified sanity check paths: %s" % paths)
 
+            # transform specified libs into simple file paths
+            if 'libs' in paths.keys():
+                for lib in paths['libs']:
+                    if 'name' in lib.keys() and 'kind' in lib.keys():
+                        if lib['kind'] in LIB_SUFFIXES.keys():
+                            # TODO: also check possible library version numbers (e.g. libawesome.so.3.0.1)
+                            libpaths = ()
+                            for dir in LIB_DIRS:
+                                for suffix in LIB_SUFFIXES[lib['kind']]:
+                                    libpaths += (os.path.join(dir, lib['name'] + suffix), )
+                            paths['files'] += [libpaths]
+                        else:
+                            raise EasyBuildError("Library kind must either be static or shared.")
+                    else:
+                        raise EasyBuildError("Libraries are specified via 'name' and 'kind'.")
+                # remove the libs
+                del paths['libs']
```

Comment:
Otherwise the following exception will be thrown:
```{.py}
if not ks == req_keys or sum(valnottypes) > 0 or sum(lenvals) == 0:
  raise EasyBuildError("Incorrect format for sanity_check_paths (should (only) have %s keys, "
                       "values should be lists (at least one non-empty)).", ','.join(req_keys))
```

This might be circumvented by implementing the library checks as a lambda in `path_keys_and_check` (similar for files and dirs) though that would make it impossible to treat the libraries as regular files (and reusing the logic for these).